Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Get Real Health.  All rights reserved.
+// MIT License
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Support;
+using Microsoft.HealthVault.Fhir.Constants;
+using HVCondition = Microsoft.HealthVault.ItemTypes.Condition;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class ConditionToHealthVault
+    {
+        public static HVCondition ToHealthVault(this Condition fhirCondition)
+        {
+            HVCondition hvCondition = fhirCondition.ToThingBase<HVCondition>();
+
+            string conditionOccurence = string.Empty;
+            var conditionExtension = fhirCondition.GetExtension(HealthVaultExtensions.Condition);
+            if (conditionExtension != null)
+            {
+                hvCondition.StopReason = conditionExtension.GetStringExtension(HealthVaultExtensions.ConditionStopReason);
+                conditionOccurence = conditionExtension.GetStringExtension(HealthVaultExtensions.ConditionOccurrence);
+            }
+
+
+            hvCondition.OnsetDate = fhirCondition.Onset.ToAproximateDateTime();
+            hvCondition.StopDate = fhirCondition.Abatement.ToAproximateDateTime();
+
+
+            if (fhirCondition.ClinicalStatus.HasValue)
+            {
+                hvCondition.Status = new ItemTypes.CodableValue(fhirCondition.ClinicalStatus.Value.ToString());
+                hvCondition.Status.Add(new ItemTypes.CodedValue(fhirCondition.ClinicalStatus.Value.ToString(), FhirCategories.Hl7Condition, HealthVaultVocabularies.Fhir, ""));
+            }
+            else if (!string.IsNullOrWhiteSpace(conditionOccurence))
+            {
+                hvCondition.Status = new ItemTypes.CodableValue(conditionOccurence);
+                hvCondition.Status.Add(new ItemTypes.CodedValue(conditionOccurence, HealthVaultVocabularies.ConditionOccurrence, HealthVaultVocabularies.Wc, "1"));
+            }
+
+            if (fhirCondition.Code.IsNullOrEmpty())
```

Comment:
>if (fhirCondition.Code.IsNullOrEmpty()) [](start = 12, length = 39)

why do we do this check this late? Move this code up. I am more inclined to return null than throw an exception. If we do not support a type, when transforming a bundle why would we like to fail the whole transformation instead of just skipping the ones we do not understand?

Reply:
@gcorvera Returning a null will leave the developer in darkness about what went wrong. 
@baiju-grh `NotSupportedException` would be better here
@jeffy-grh We need to standardize and document possible exceptions.